lib/Conversion/ImportVerilog/ImportVerilogInternals.h

```diff
@@ -64,6 +64,7 @@ struct Context {
 
   // Convert an expression AST node to MLIR ops.
   Value convertExpression(const slang::ast::Expression &expr);
+  Value &convertRHSExpression(const slang::ast::Expression &expr);
```

Comment:
Yes I totally agree! In the software world these are usually called L-values and R-values, referring to whether the expression appeared on the **L**eft or **R**ight side of an assignment. This only talks about assignments, not other expressions that have an LHS and RHS operand.

The reason is that an addition like `a + b` just needs values for `a` and `b` that it can add. But an assignment like `c = d` requires _a pointer to where `c`'s new value can be stored_ and the value for `d`. This is what lvalue vs. rvalue lowerings handle: lvalue lowering gives you _an assignable version_ of the expression, while rvalue lowering gives you _the value_ of the exprression. And lvalue lowering usually only supports a small subset of all possible expressions , e.g. only variable names, array indexing, and struct fields. While rvalue lowering supports all possible expressions.

So I would call these specifically `convertLvalueExpression` and `convertRvalueExpression`, to differentiate lvalue/rvalues in assignments from LHS/RHS operands in arbitrary binary expressions. `convertLvalueExpression` would then only support variable names in the beginning and would just return the `ref<T>` of that variable. And in `convertRvalueExpression`, when you see a variable name, you'd actually `moore.read` that variable to turn it into a `T`.